src/include/switch_core.h

```diff
@@ -1325,6 +1325,13 @@ SWITCH_DECLARE(uint32_t) switch_core_session_flush_private_events(switch_core_se
 */
 SWITCH_DECLARE(uint32_t) switch_core_session_flush_and_publish_private_events(switch_core_session_t *session);
 
+/*!
+  \brief Flush only private media events queue of a session and publish COMMAND_INTERRUPTED event
+  \param session the session to flush
+  \return SWITCH_STATUS_SUCCESS if the events have been flushed
+*/
+SWITCH_DECLARE(uint32_t) switch_core_session_flush_and_publish_private_media_events(switch_core_session_t *session, const char *playback_cmd, 
```

Comment:
This is not flexible to handle new commands to pass in this function. I would suggest to update the function parameter
```
SWITCH_DECLARE(uint32_t) switch_core_session_flush_and_publish_private_media_events(switch_core_session_t *session, char *commands, int commandlen) 
```
In your code you will have to iterate that function and validate if it matches in the command list
```
int index = 0;
int matches = 0;
....
for(index = 0; index < commandlen; ++index) {
  if (!strcasecmp(app_name, commands[index])) {
    matches = 1;
    break;
  }
}
if (matches) {
  if (switch_event_create(&interrupt_event, SWITCH_EVENT_COMMAND_INTERRUPTED) == SWITCH_STATUS_SUCCESS) {
     ...
  }
}
```

Reply:
I actually thought of this. and I noticed this would also modify the complexity of the function to n^2 which I think might not also be a good idea.

So the question is, is it really worth it to trade off performance for scalability? 

Anyways, I'll see how to improve that part of the code

Reply:
My bad. Just checked the sample you shared properly and I see this is just a 2n complexity. 

I'll look into it. Thanks.